### PR TITLE
Comparison of related model to use pk instead of id

### LINKED ISSUE #4180: Compare with previous version fails when model has custom primary key field
### Issue Summary

Compare with previous version fails when model has custom primary key field. `ChildRelationComparison.get_mapping` uses `id` instead of `pk`, causing any models with custom primary keys to raise AttributeError.

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Edit models.py as follows...

```
# -*- coding: utf-8 -*-
from __future__ import absolute_import, division, print_function, unicode_literals

from django.db import models
from modelcluster.fields import ParentalKey
from wagtail.wagtailadmin.edit_handlers import FieldPanel, InlinePanel
from wagtail.wagtailcore.models import Page


class HomePage(Page):
    content_panels = Page.content_panels + [
        InlinePanel('related_models', label='Related models'),
    ]


class RelatedModel(models.Model):
    custom_id = models.AutoField(primary_key=True)
    home_page = ParentalKey('home.HomePage', related_name='related_models')
    value = models.IntegerField()

    panels = [
        FieldPanel('value'),
    ]
```
3. `makemigrations`
4. `migrate`
5. `createsuperuser`
5. `runserver`
6. login to /admin
7. edit home page
8. add a "related model" and press save draft
9. add a "related model" and press publish
10. re-edit home page
11. click revisions
12. click "compare with previous revision"

I expected it to use the correct primary key field. Code in `ChildRelationComparison.get_mapping` should use `pk` instead of `id`.

### Technical details

* Python version: 2.7.10
* Django version: 1.11.9
* Wagtail version: 1.13.1
* Browser version: whatsmybrowser.org/b/XXQTSVL

### PR DESCRIPTION
**Pull request by lb-**
Resolves #4180

Comparing related models now uses .pk instead of .id when comparing if the related objects have changed or not.

Added tests for id (pk) comparisons, no docs to update as this enhancement reflects expected behaviour.

### CODE REVIEW

**Comment by lb-** [comment]
@gasman could you check through this when you get the chance

File: wagtail/tests/testapp/models.py [hunk 1]
```diff
@@ -295,6 +297,18 @@ class EventPage(Page):
 ]
 
 
+class HeadCountRelatedModelUsingPK(models.Model):
+    """Related model that uses a custom primary key (pk) not id"""
+    custom_id = models.AutoField(primary_key=True)
```
Comment (BertrandBordage) [inline wagtail/tests/testapp/models.py]: It would test more edge cases to use a non-integer primary key, here.

Comment (webit-mdowney) [reply wagtail/tests/testapp/models.py]: `UUIDField(primary_key=True, default=uuid.uuid1)`

**Comment by lb-** [comment]
@BertrandBordage @gasman this has been merged but I am good to add a new PR for an additional test using a uuid primary key as well if needed.

**Pull request merged**